torchgen/api/python.py

```diff
@@ -1002,13 +1002,15 @@ def returns_named_tuple_pyi(signature: PythonSignature) -> Optional[Tuple[str, s
         namedtuple_def_lines.extend(
             f"    {name}: {typ}" for name, typ in zip(field_names, python_returns)
         )
+        namedtuple_def_lines.append("    def __init__(self, sequence: Sequence): ...")
```

Comment:
It looks to me like `NamedTuple` intentionally doesn't have an overload to build from a sequence. It even has a seperate method `_make` which does extra checks on top of `__new__`. Perhaps just use `NamedTuple(*sequence)` in your PR.

Reply:
Hrm okay so this typing is a bit of a lie then, but I can't see any `typing.structseq` or similar so I guess this is good enough. It would be good to add a comment to that effect though.

Reply:
cc @Fidget-Spinner who maintains `typing` IIRC

Reply:
cc @Skylion007 

We could also inherit from NamedTuple and then lie and say that `torch.return_types.frexp(torch.zeros(2), torch.zeros(2, dtype=torch.int32))` works even if it doesn't via an `overrides` (a hack going in a different direction)

Reply:
I used to maintain `typing` but since retired from that. However, would it be possible to inherit from [collections.namedtuple](https://docs.python.org/3/library/collections.html#collections.namedtuple) directly? The `namedtuple._make(iterable)` signature is exactly like the limitations structseq has in its init. The type checker should be infer the namedtuple, the downside is the exact types of its fields might not be inferred.

If we can't do that, then this seems like a good workaround IMO.

Reply:
Directly using `collections.namedtuple` doesn't give us a way of typing the values like `mantissa` and `exponent`. I tried fixing it, but ended up with something close to the original PR.

Reply:
> Hrm okay so this typing is a bit of a lie then, but I can't see any `typing.structseq` or similar so I guess this is good enough. It would be good to add a comment to that effect though.

FYI, there is a dummy type stub in [python/typeshed](https://github.com/python/typeshed):

https://github.com/python/typeshed/blob/a2095002e446bd0e20f8a469a14e271cd6cc6ad9/stdlib/_typeshed/__init__.pyi#L299-L314

```python
# This is an internal CPython type that is like, but subtly different from, a NamedTuple
# Subclasses of this type are found in multiple modules.
# In typeshed, `structseq` is only ever used as a mixin in combination with a fixed-length `Tuple`
# See discussion at #6546 & #6560
# `structseq` classes are unsubclassable, so are all decorated with `@final`.
class structseq(Generic[_T_co]):
    n_fields: Final[int]
    n_unnamed_fields: Final[int]
    n_sequence_fields: Final[int]
    # The first parameter will generally only take an iterable of a specific length.
    # E.g. `os.uname_result` takes any iterable of length exactly 5.
    #
    # The second parameter will accept a dict of any kind without raising an exception,
    # but only has any meaning if you supply it a dict where the keys are strings.
    # https://github.com/python/typeshed/pull/6560#discussion_r767149830
    def __new__(cls: type[Self], sequence: Iterable[_T_co], dict: dict[str, Any] = ...) -> Self: ...
```